log4j-core/src/main/java/org/apache/logging/log4j/core/layout/Rfc5424Layout.java

```diff
@@ -642,80 +642,128 @@ public String toString() {
-                required,
-                StandardCharsets.UTF_8,
-                exceptionPattern,
-                useTlsMessageFormat,
-                loggerFields);
-    }
-
-    public static class Rfc5424LayoutBuilder {
-        private Configuration config;
+        return newBuilder()
+                .setConfiguration(config)
+                .setFacility(facility)
+                .setId(id)
+                .setEin(String.valueOf(enterpriseNumber))
+                .setIncludeMDC(includeMDC)
+                .setIncludeNL(newLine)
+                .setEscapeNL(escapeNL)
+                .setMdcId(mdcId)
+                .setMdcPrefix(mdcPrefix)
+                .setEventPrefix(eventPrefix)
+                .setAppName(appName)
+                .setMessageId(msgId)
+                .setExcludes(excludes)
+                .setIncludes(includes)
+                .setRequired(required)
+                .setCharset(StandardCharsets.UTF_8)
+                .setExceptionPattern(exceptionPattern)
+                .setUseTLSMessageFormat(useTlsMessageFormat)
+                .setLoggerFields(loggerFields)
+                .build();
+    }
+
+    @PluginBuilderFactory
+    public static Rfc5424LayoutBuilder newBuilder() {
+        return new Rfc5424LayoutBuilder();
+    }
+
+    public static class Rfc5424LayoutBuilder extends AbstractStringLayout.Builder<Rfc5424LayoutBuilder>
+            implements org.apache.logging.log4j.core.util.Builder<Rfc5424Layout> {
+
+        @PluginBuilderAttribute
         private Facility facility = Facility.LOCAL0;
+
+        @PluginBuilderAttribute
         private String id;
+
+        @PluginBuilderAttribute
         private String ein = String.valueOf(DEFAULT_ENTERPRISE_NUMBER);
+
+        @PluginBuilderAttribute
+        private Integer enterpriseNumber;
+
+        @PluginBuilderAttribute
         private boolean includeMDC = true;
+
+        @PluginBuilderAttribute
         private boolean includeNL;
+
+        @PluginBuilderAttribute
         private String escapeNL;
+
+        @PluginBuilderAttribute
         private String mdcId = DEFAULT_MDCID;
+
+        @PluginBuilderAttribute
         private String mdcPrefix;
+
+        @PluginBuilderAttribute
         private String eventPrefix;
+
+        @PluginBuilderAttribute
         private String appName;
+
+        @PluginBuilderAttribute
         private String messageId;
+
+        @PluginBuilderAttribute
         private String excludes;
+
+        @PluginBuilderAttribute
         private String includes;
+
+        @PluginBuilderAttribute
         private String required;
-        private Charset charset;
+
+        @PluginBuilderAttribute
         private String exceptionPattern;
+
+        @PluginBuilderAttribute
         private boolean useTLSMessageFormat;
+
+        /**
+         * If {@code true}, the FQDN of the current host will be used.
+         */
+        @PluginBuilderAttribute
+        private boolean useFqdn = false;
```

Comment:
Given that enabling this fixes our RFC5424 compliance, shouldn't this feature be a default instead of opt-in? I am aware that this will be a breaking change, yet, I'd favor an approach as follows:

1. Change the default
2. If we receive a bug report, introduce the flag as an opt-out (contrary to the current opt-in)

Reply:
Excellent idea!

Fixed in https://github.com/apache/logging-log4j2/pull/2853/commits/082f3683dce1530c5d735ea413efecbe3e1fa320